explore_data/cluster_by_accessory_gene.py

```diff
@@ -0,0 +1,609 @@
+fig4 += pn.theme(
+    legend_title_align="center",
+    plot_background=pn.element_rect(fill="white"),
+    legend_key=pn.element_rect(fill="white", colour="white"),
+    legend_title=pn.element_text(family="sans-serif", size=15),
+    legend_text=pn.element_text(family="sans-serif", size=12),
+    plot_title=pn.element_text(family="sans-serif", size=15),
+    axis_text=pn.element_text(family="sans-serif", size=10),
+    axis_title=pn.element_text(family="sans-serif", size=12),
+)
+fig4 += pn.guides(colour=pn.guide_legend(override_aes={"alpha": 1}))
+
+
+print(fig4)
+# -
+
+# **Note:**
+# We can compare TPM between PAO1 and PA14 if they have similar total number of reads mapped. So perhaps we can compare PAO1 and PA14 compendia. See Georgia’s [google doc](https://docs.google.com/document/d/1OnvJNVkhK5ATnhHaeSmz8Ze_Iw9ixXhfSxAL_BSqbuM/edit)
+
+# **Takeaway:**
+# * This is a positive control that PAO1 annotated samples have higher median expression of PAO1-only genes compared to PA14-only genes. And similarly, PA14 annotated samples have higher median expression of PA14-only genes compared to PAO1-only genes. In other words, we expect that PA14-only genes will have either 0 or very low values in PAO1 samples and vice versa.
+#
+# * This result also shows that we can anticipate a very clear binning of our samples into PAO1 and PA14 if we use mapping rates.
+#
+# * Note: The NA strains are those where the strain information was not available in the metadata. By a quick manual spot check it looks like a bunch were clinical isolates (which is good since these NA seem to cluster with other clinical isolates).
+
+# ### How many samples have a very low or 0 expression?
+
+# +
+# Total number of PAO1 samples
+num_pao1_samples = (pao1_pa14_acc_expression_label["Strain type_pao1"] == "PAO1").sum()
+print(f"Number of PAO1 samples is {num_pao1_samples}")
+
+# Total number of PA14 samples
+num_pa14_samples = (pao1_pa14_acc_expression_label["Strain type_pao1"] == "PA14").sum()
+print(f"Number of PA14 samples if {num_pa14_samples}")
+print("\n\n")
+
+# Of those samples, how many have o or low expression?
+threshold = 50
+pao1_sample_ids = pao1_pa14_acc_expression_label["Strain type_pao1"] == "PAO1"
+pao1_acc_expression_label = pao1_pa14_acc_expression_label.loc[pao1_sample_ids]
+pao1_core_expression_label = pao1_pa14_core_expression_label.loc[pao1_sample_ids]
+
+print("----- Matching sample and reference statistics -------")
+pao1_acc_expression_low = pao1_acc_expression_label[
+    pao1_acc_expression_label["median acc expression_pao1"] == 0
+]
+print(
+    f"Proportion of PAO1 samples with 0 median PAO1-only expression: {pao1_acc_expression_low.shape[0]/num_pao1_samples}"
+)
+pao1_acc_expression_low = pao1_acc_expression_label[
+    pao1_acc_expression_label["median acc expression_pao1"] < threshold
+]
+print(
+    f"Proportion of PAO1 samples with median PAO1-only expression < {threshold}: {pao1_acc_expression_low.shape[0]/num_pao1_samples}"
+)
```

Comment:
Why are you interested in the number of PA01 samples where the expression is below a threshold?

Reply:
This was more for me to get a better sense for where the expression is distributed, so it looks like most samples have a median expression < 50 TPM. This would allow be to quantify the distribution a bit